app/models/shipit/user.rb

```diff
@@ -123,5 +129,20 @@ def identify_renamed_user!
     rescue Octokit::NotFound
       false
     end
+
+    def org_email?(email_address)
+      return false if email_address.blank?
+
+      org_domains = Shipit.preferred_org_emails
+      return true if org_domains.empty? || org_domains.all?(&:blank?)
+
+      org_domains.any? { |domain| email_address.end_with?("@#{domain}") }
+    end
+
+    def filter_org_emails(github_email_records)
+      org_email_records = github_email_records.select { |email_record| org_email?(email_record.email) }
+      email_record = org_email_records.find(-> { github_email_records.first }, &:primary)
```

Comment:
🤔I can't tell what this does, you're going to have to humour me - why do we go back to `github_email_records` here?

Reply:
Whoops, that was a typo from when I was renaming these fields. I'll fix and add a test to cover the case (which is point 6 on the https://github.com/Shopify/shipit/issues/775).